core-plugins/src/test/java/co/cask/hydrator/plugin/batch/joiner/JoinerTestRun.java

```diff
@@ -0,0 +1,264 @@
+                                 BatchJoiner.PLUGIN_TYPE,
+                                 ImmutableMap.of(
+                                   "joinKeys", "film.film_id=filmActor.film_id=filmCategory.film_id&" +
+                                     "film.film_name=filmActor.film_name=filmCategory.film_name",
+                                   "selectedFields", GSON.toJson(selectedFields),
+                                   "requiredInputs", "film,filmActor,filmCategory"),
+                                 null));
+
+    // output schema sorted by input stage names
+    Schema outputSchema = Schema.recordOf(
+      "joined",
+      Schema.Field.of("film_id", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("film_name", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("renamed_actor", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("renamed_category", Schema.of(Schema.Type.STRING)));
+
+    ETLStage joinSinkStage = new ETLStage(
+      "sink", new ETLPlugin("TPFSAvro", BatchSink.PLUGIN_TYPE,
+                            ImmutableMap.of(Properties.TimePartitionedFileSetDataset.SCHEMA, outputSchema.toString(),
+                                            Properties.TimePartitionedFileSetDataset.TPFS_NAME, joinedDatasetName),
+                            null));
+
+    ETLBatchConfig config = ETLBatchConfig.builder("* * * * *")
+      .addStage(filmStage)
+      .addStage(filmActorStage)
+      .addStage(filmCategoryStage)
+      .addStage(joinStage)
+      .addStage(joinSinkStage)
+      .addConnection(filmStage.getName(), joinStage.getName())
+      .addConnection(filmActorStage.getName(), joinStage.getName())
+      .addConnection(filmCategoryStage.getName(), joinStage.getName())
+      .addConnection(joinStage.getName(), joinSinkStage.getName())
+      .build();
+    AppRequest<ETLBatchConfig> request = new AppRequest<>(DATAPIPELINE_ARTIFACT, config);
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "joiner-test");
+    ApplicationManager appManager = deployApplication(appId, request);
+
+    // write input data
+    // 1: 1, matrix
+    // 2: 2, equilibrium
+    // 3: 3, avatar
+    DataSetManager<Table> filmManager = getDataset(filmDatasetName);
+    Table filmTable = filmManager.get();
+    Put put = new Put(Bytes.toBytes(1));
+    put.add("film_id", "1");
+    put.add("film_name", "matrix");
+    filmTable.put(put);
+
+    put = new Put(Bytes.toBytes(2));
+    put.add("film_id", "2");
+    put.add("film_name", "equilibrium");
+    filmTable.put(put);
+
+    put = new Put(Bytes.toBytes(3));
+    put.add("film_id", "3");
+    put.add("film_name", "avatar");
+    filmTable.put(put);
+    filmManager.flush();
+
+    // 1: 1, matrix, alex
+    // 2: 1, matrix, bob
+    // 3: 2, equilibrium, cathie
+    // 4: 3, avatar, samuel
+    DataSetManager<Table> filmActorManager = getDataset(filmActorDatasetName);
+    Table filmActorTable = filmActorManager.get();
+    put = new Put(Bytes.toBytes(1));
```

Comment:
private methods for creating these `Puts` would make the code more readable/clean